src/tasks/minions/monsterActivity.ts

```diff
@@ -304,7 +315,25 @@ export const monsterTask: MinionTask = {
 			const quantityLeft = Math.max(0, isOnTaskResult.currentTask!.quantity_remaining - effectiveSlayed);
 
 			thisTripFinishesTask = quantityLeft === 0;
-			if (thisTripFinishesTask) {
+			if (thisTripFinishesTask && isOnTaskResult.slayerMaster.id === 8) {
```

Comment:
prefer `const isUsingKristila = isOnTaskResult.slayerMaster.id === 8` so this is more clear what its doing.